GVFS/GVFS.Common/Git/GitAuthentication.cs

```diff
@@ -45,6 +46,16 @@ public void ConfirmCredentialsWorked(string usedCredential)
                 {
                     this.numberOfAttempts = 0;
                     this.lastAuthAttempt = DateTime.MinValue;
+
+                    // Tell Git to store the valid credential if we haven't already
+                    // done so for this cached credential.
+                    string username;
+                    string password;
+                    if (!this.storedCachedAuthString && this.TryParseCachedAuthString(out username, out password))

```

Comment:
Using a `SecureString` is usually more hassle than it's worth. A few reasons not to bother IMO:
1. `SecureString` isn't actually encrypted on Unix systems, only on Windows.
2. `SecureString` lives in unmanaged memory as a `BSTR`, meaning you need to jump through hoops to get access to it. I mean, this is kind of the point (to prevent accidental usage of the string in normal code).
3. If you're able to read managed memory to get access to a normal `System.String` then you can also just watch what GCM and Git are passing back and forth over standard input.
4. Anyone can just ask GCM to dump out their PATs knowing only the repo URL without any restrictions other than running as the current user (unless the user chooses to lock their own login keychain on Mac; Windows has no such protection).

Reply:
https://docs.microsoft.com/en-us/dotnet/api/system.security.securestring?view=netframework-4.7.2#how-secure-is-securestring

> Overall, SecureString is more secure than String because it limits the exposure of sensitive string data. However, those strings may still be exposed to any process or operation that has access to raw memory, such as a malicious process running on the host computer, a process dump, or a user-viewable swap file. **Instead of using SecureString to protect passwords, the recommended alternative is to use an opaque handle to credentials that are stored outside of the process\*.**

\* You could read the last one as only interfacing via GCM which reads/writes the credentials at rest from the OS credential store.